src/FitterBase.cc

```diff
@@ -299,5 +305,18 @@ void FitterBase::saveChi2Contributions(std::string const &baseName) const {
     saveChi2RefContributions(refTuple);
 }
 
+float FitterBase::_lineSearch(Eigen::VectorXd const &delta) {
+    auto func = [this, &delta](double scale) {
+        offsetParams(scale * delta);
+        auto chi2 = computeChi2();
+        // reset the system to where it was before offsetting.
+        offsetParams(-scale * delta);
```

Comment:
I'm not a great fan of this setting/resetting, or the state changing in general for something that could be a `const` function, but I assume it is unavoidable?

Reply:
It is necessary to change the state in order to compute the chi2 at the new position.

Another option would be to make a copy of the model and offset the copy, but the model is very large, so allocate/copy/add is probably of comparable speed to add/subtract (in place) as implemented, but uses more memory and requires implementing copy for the models (which have many shared pointers to the underlying transforms, each of which would have to be deep copied). Also, any numerical imprecision induced by the state change should be well below the overall precision desired by the model.